tests/integration/suite/actors/metadata/actor_client.go

```diff
@@ -0,0 +1,101 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metadata
+
+import (
+	"context"
+	"net/http"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/dapr/dapr/tests/integration/framework"
+	"github.com/dapr/dapr/tests/integration/framework/process/daprd"
+	prochttp "github.com/dapr/dapr/tests/integration/framework/process/http"
+	"github.com/dapr/dapr/tests/integration/framework/process/placement"
+	"github.com/dapr/dapr/tests/integration/framework/util"
+	"github.com/dapr/dapr/tests/integration/suite"
+)
+
+func init() {
+	suite.Register(new(metadataActorClient))
+}
+
+// metadataActorClient tests the response of the metadata API for a healthy actor client.
+type metadataActorClient struct {
+	daprd       *daprd.Daprd
+	place       *placement.Placement
+	blockConfig chan struct{}
+}
+
+func (m *metadataActorClient) Setup(t *testing.T) []framework.Option {
+	m.blockConfig = make(chan struct{})
+
+	handler := http.NewServeMux()
+	handler.HandleFunc("/dapr/config", func(w http.ResponseWriter, r *http.Request) {
+		<-m.blockConfig
+		w.Write([]byte(`{"entities": []}`))
+	})
+	handler.HandleFunc("/healthz", func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusOK)
+	})
+	handler.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+		w.Write([]byte(`OK`))
+	})
+
+	srv := prochttp.New(t, prochttp.WithHandler(handler))
+	m.place = placement.New(t)
+	m.daprd = daprd.New(t,
+		daprd.WithPlacementAddresses("localhost:"+strconv.Itoa(m.place.Port())),
+		daprd.WithAppProtocol("http"),
+		daprd.WithAppPort(srv.Port()),
+		daprd.WithLogLevel("info"), // Daprd is super noisy in debug mode when connecting to placement.
+	)
+
+	return []framework.Option{
+		framework.WithProcesses(m.place, srv, m.daprd),
+	}
+}
+
+func (m *metadataActorClient) Run(t *testing.T, ctx context.Context) {
+	// Test an app that is an actor client (no actor state store configured)
+	// 1. Assert that status is "INITIALIZING" before /dapr/config is called
+	// 2. After init is done, status is "RUNNING", hostReady is "false", placement reports a connection, and hosted actors are empty
+
+	m.place.WaitUntilRunning(t, ctx)
+	m.daprd.WaitUntilTCPReady(t, ctx)
+
+	client := util.HTTPClient(t)
+
+	// Before initialization
+	res := getMetadata(t, ctx, client, m.daprd.HTTPPort())
+	require.False(t, t.Failed())
+	assert.Equal(t, "INITIALIZING", res.ActorRuntime.RuntimeStatus)
+	assert.False(t, res.ActorRuntime.HostReady)
+	assert.Empty(t, res.ActorRuntime.Placement)
+	assert.Empty(t, res.ActorRuntime.ActiveActors)
+
+	// Complete init
+	close(m.blockConfig)
+	assert.EventuallyWithT(t, func(t *assert.CollectT) {
+		res := getMetadata(t, ctx, client, m.daprd.HTTPPort())
+		assert.Equal(t, "RUNNING", res.ActorRuntime.RuntimeStatus)
+		assert.False(t, res.ActorRuntime.HostReady)
```

Comment:
clarification question: Based on https://github.com/dapr/dapr/pull/7040/files#diff-ab620a0df90b230287b642dd2222bc40d6092a80d11092cd654da649de77d85fR137 

When app is healthy and placement is connected, host must be ready ... 
So why is that not the case here?

Reply:
See my response on the other comment below

Reply:
Ok I see now, that a state store is not configured for this test which I see in the `actor_host.go`. That would then make the host not ready.